lib/snapshot.js

```diff
@@ -0,0 +1,271 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const zlib = require('zlib');
+const md5Hex = require('md5-hex');
+const mkdirp = require('mkdirp');
+const kathryn = require('kathryn');
+
+const HASH_BYTES = 16; // A md5 hash is 16 bytes long.
+
+// Increment if encoding layout or Kathryn serialization versions change. Previous AVA versions will not be able to
+// decode buffers generated by a newer version, so changing this value will require a major version bump of AVA itself.
+// The version is encoded as an unsigned 8 bit integer. If it ever reaches 255 it *must* be encoded as a 16 bit integer
+// instead.
```

Comment:
So let's go with 16 bit from the start then? Then we don't have to document such limitation and never have to worry about it.